src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -423,6 +433,7 @@ private bool ShouldBeInStateNodesMode(Snapshot best)
             bool hasFastSyncBeenActive = best.Header >= PivotNumber;
             bool hasAnyPostPivotPeer = AnyPostPivotPeerKnown(best.PeerBlock);
             bool notInFastSync = !best.IsInFastSync;
+            bool notNeedToWaitForHeaders = NotNeedToWaitForHeaders;
```

Comment:
Is it needed for the State Sync in ShouldBeInStateNodesMode ?
Why can't we start immediately ?

Reply:
we need to DL and validate Head header in order to know which state root to download